controller/konnect/konnectextension_controller_utils.go

```diff
@@ -134,6 +138,72 @@ func (r *KonnectExtensionReconciler) getKonnectControlPlane(
 	return konnectCP, ctrl.Result{}, err
 }
 
+// ensureExtendablesReferencesInStatus ensures that the KonnectExtension references to DataPlane and ControlPlane are up-to-date.
+// Only DataPlanes and ControlPlanes with the condition KonnectExtensionApplied=True are added to the status.
+func (r *KonnectExtensionReconciler) ensureExtendablesReferencesInStatus(
+	ctx context.Context,
+	ext *konnectv1alpha1.KonnectExtension,
+	dps operatorv1beta1.DataPlaneList,
+	cps operatorv1beta1.ControlPlaneList,
+) (ctrl.Result, error) {
+	sortRefs := func(refs []commonv1alpha1.NamespacedRef) {
+		refToStr := func(ref commonv1alpha1.NamespacedRef) string {
+			// We can safely assume that the namespace is not nil, as we fill it when mapping refs.
+			return fmt.Sprintf("%s/%s", *ref.Namespace, ref.Name)
+		}
+		sort.Slice(refs, func(i, j int) bool {
+			return refToStr(refs[i]) < refToStr(refs[j])
+		})
+	}
+	hasExetensionAppliedCondition := func(conditions []metav1.Condition) bool {
+		return lo.ContainsBy(conditions, func(cond metav1.Condition) bool {
+			return cond.Type == string(konnect.KonnectExtensionAppliedType) &&
+				cond.Status == metav1.ConditionTrue
+		})
+	}
+
+	extOld := ext.DeepCopy()
+
+	// Ensure DataPlaneRefs are up-to-date.
+	var dpRefs []commonv1alpha1.NamespacedRef
+	for _, dp := range dps.Items {
+		// Only add DataPlanes with the KonnectExtensionApplied condition set to true.
+		if !hasExetensionAppliedCondition(dp.Status.Conditions) {
+			continue
+		}
+		dpRefs = append(dpRefs, commonv1alpha1.NamespacedRef{
+			Name:      dp.Name,
+			Namespace: &dp.Namespace,
+		})
+	}
+	sortRefs(dpRefs)
+	ext.Status.DataPlaneRefs = dpRefs
+
+	// Ensure ControlPlaneRefs are up-to-date.
+	var cpRefs []commonv1alpha1.NamespacedRef
+	for _, cp := range cps.Items {
+		// Only add ControlPlanes with the KonnectExtensionApplied condition set to true.
+		if !hasExetensionAppliedCondition(cp.Status.Conditions) {
+			continue
+		}
+		cpRefs = append(cpRefs, commonv1alpha1.NamespacedRef{
+			Name:      cp.Name,
+			Namespace: &cp.Namespace,
+		})
+	}
+	sortRefs(cpRefs)
+	ext.Status.ControlPlaneRefs = cpRefs
+
+	if shouldUpdate := !cmp.Equal(ext.Status, extOld.Status); !shouldUpdate {
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.Client.Status().Update(ctx, ext); err != nil {
```

Comment:
Can we handle conflicts with `k8serrors.IsConflict` and not return an `err` then but a requeue set to `true`?